### PR TITLE
JS-299 List nodes that need to be supported in TS

### PR DESCRIPTION
**Pull request by ericmorand-sonarsource**
Fixes JS-299

### CODE REVIEW

File: packages/jsts/src/parsers/ast.ts [hunk 1]
```diff
@@ -636,125 +737,119 @@ export function visitNode(node: estree.BaseNodeWithoutComments | undefined | nul
...
-    } else {
-      return {
-        expression: visitNode(node.expression),
-      };
-    }
+  function visitExpressionStatement(node: TSESTree.ExpressionStatement) {
+    return {
+      expression: visitNode(node.expression),
+      directive: node.directive,
+    };
```
Comment (ilia-kebets-sonarsource) [inline packages/jsts/src/parsers/ast.ts:841]: this change will have repercussions

**Review by ilia-kebets-sonarsource — CHANGES_REQUESTED** [review]
Can you adapt the changes to expression statement serializing in the deserialization?

File: packages/jsts/src/parsers/ast.ts (hunk 1 above)
Comment (ericmorand-sonarsource) [reply packages/jsts/src/parsers/ast.ts:841]: I'm not sure I understand the issue. If `node.directive` is undefined, `directive` will be undefined and will be serialized as such, just like if it was not present in the seralizable output, isn't it?

Comment (ericmorand-sonarsource) [reply packages/jsts/src/parsers/ast.ts:841]: I'll revert the change anyway.

**Review by ilia-kebets-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**